src/Database/DBStructure.php

```diff
@@ -48,6 +48,22 @@ class DBStructure
 	 */
 	private static $definition = [];
 
+	/**
+	 * Set a database version to trigger update functions
+	 *
+	 * @param string $version
+	 * @return void
+	 */
+	public static function setDatabaseVersion(string $version)
```

Comment:
What would happen with a non numeric version string then?

Reply:
It will be cast as an integer, and it it is impossible (like with `null`) then it throws an exception. String representation of integers are fine. If we aren't using leading zeroes, then we don't need strings.

Reply:
From a user's perspective I prefer some readable and understandable message that tells the user that the input is not numeric - instead of some unreadable exception.

Reply:
What user are you talking about? Friendica user? Admin? Developer? The first doesn't see these errors, the second should report it to the third who is perfectly capable to understand the exception message.

These exceptions have allowed us to find spots where the data wasn't present like we expected. So it defaulted to a falsy value when we didn't have the information. Type-hinting helps making sure we have the right data.

Reply:
Exceptions aren't meant to be thrown to the user. They contains information for debugging and therefore are meant to be logged for devs.

But the reason behind the type hints is that the method itself has the constraint to disallow wrong types. Therefore it is the problem of the caller of the db update method to catch wrongly determined versions, not the method itself.

So I'd add the type hint and add a try-catch block around it. A more convenient way would be to surround the CLI commands itself with a try catch block so at the most outer functionality, and therr we catch every error and map it to the CLI output..